pkg/system/rm.go

```diff
@@ -63,12 +63,8 @@ func EnsureRemoveAll(dir string) error {
 			return err
 		}
 
-		if mounted, _ := mount.Mounted(pe.Path); mounted {
-			if e := mount.Unmount(pe.Path); e != nil {
-				if mounted, _ := mount.Mounted(pe.Path); mounted {
-					return errors.Wrapf(e, "error while removing %s", dir)
-				}
-			}
+		if e := mount.Unmount(pe.Path); e != nil {
+			return errors.Wrapf(e, "error while removing %s", dir)
```

Comment:
In the existing code, the Unmount command would not be called, and we would not get an error if the code was not Mounted,  Why not check if the error is EINVAL, and then check if it is Mounted, and not return an error.
Then you are fast in the default path, and continue the functionality of not returning EINVAL errors.

Reply:
`mount.Unmount()` already ignores `EINVAL`